src/Data/Nat/Properties.agda

```diff
@@ -822,6 +822,21 @@ i^j≡1⇒j≡0∨i≡1 : ∀ i j → i ^ j ≡ 1 → j ≡ 0 ⊎ i ≡ 1
 i^j≡1⇒j≡0∨i≡1 i zero    _  = inj₁ refl
 i^j≡1⇒j≡0∨i≡1 i (suc j) eq = inj₂ (i*j≡1⇒i≡1 i (i ^ j) eq)
 
+------------------------------------------------------------------------
+-- Properties of _≤‴_
+
+≤‴⇒≤″ : ∀{m n} → m ≤‴ n → m ≤″ n
+≤‴⇒≤″ {m = m} ≤‴-refl = less-than-or-equal {k = 0} (+-identityʳ m)
+≤‴⇒≤″ {m = m} (≤‴-step x) = less-than-or-equal (trans (+-suc m _) (_≤″_.proof ind)) where
+  ind = ≤‴⇒≤″ x
+
+≤″⇒≤‴ : ∀{m n} → m ≤″ n → m ≤‴ n
+≤″⇒≤‴ (less-than-or-equal {k} proof) = h proof where
```

Comment:
I think this proof simplifies by pattern matching on `proof` to `refl`?

Reply:
I couldn't make it simpler. Do you have something in mind?

Reply:
You could do something like this:

```agda
m≤‴n+m : ∀ {m} n → m ≤‴ n + m
m≤‴n+m zero = ≤‴-refl
m≤‴n+m {m} (suc n) rewrite sym (+-suc n m) = ≤‴-step (m≤‴n+m n)

m≤‴m+n : ∀ {m} n → m ≤‴ m + n
m≤‴m+n {m} n rewrite +-comm m n = m≤‴n+m n

≤″⇒≤‴ : ∀{m n} → m ≤″ n → m ≤‴ n
≤″⇒≤‴ {m} (less-than-or-equal {k} refl) = m≤‴m+n k
```

Reply:
I try not to use ```with``` or substitution in the inductive step. For similar reasons, we have [cast](https://github.com/agda/agda-stdlib/blob/64c90354189b9d497310197cccb989d30caa715e/src/Data/Fin/Base.agda#L47) for ```Fin```.